### PR TITLE
Add compatibility matrix

### PR DESCRIPTION
**Pull request by sfoo-iohk**
This page can be used for release tracking and eventually version skew policies as well as EOL support.

### CODE REVIEW

File: docs/platform-and-architecture/versioning-policy.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+title: Versioning policy
+sidebar_position: 8
+---
+
+## Support and release cadence
+
+At the moment, Marlowe does not strictly adhere to [Semantic Versioning](https://semver.org/). Prior to a `v1` release, changes may be breaking and backporting future fixes is to be determined.
```
Comment (joseph-fajen) [inline docs/platform-and-architecture/versioning-policy.md:8]: To make the link more obvious, please make it bold.

File: docs/platform-and-architecture/versioning-policy.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+Releases are not operating at a regular cadence yet although critical bug fixes will typically lead to faster releases.
```
Comment (joseph-fajen) [inline docs/platform-and-architecture/versioning-policy.md:10]: Please add a comma after "yet"

**Review by joseph-fajen — CHANGES_REQUESTED** [review]
Could you make one or two statements that make the practical impact of this matrix more explicit and abundantly clear? It may be expected to be evident to the intended audience. But speaking for myself, I'm not sure what I should do with this info if I want to use Marlowe. Please clarify the purpose of this page and why it is needed.

File: docs/platform-and-architecture/versioning-policy.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+The jump in Cardano node versions represent the version number matching the protocol version of Cardano.
+
+`✗` - Incompatible
+
+`✓` - Compatible
+
+`?` - Untested
+
+:::
```
[the remark this answers is not in the document]
Comment (sfoo-iohk) [reply docs/platform-and-architecture/versioning-policy.md]: fyi, this is Marlowe parity of https://docs.cardano.org/tools/comp-matrix/

If a user abstracted away these pieces through Demeter, etc, this document is not relevant to them

**Review by joseph-fajen — APPROVED** [review]

File: docs/platform-and-architecture/versioning-policy.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+| [0.0.11.0](https://github.com/input-output-hk/marlowe-cardano/tree/marlowe-cli%40v0.0.11.0) | June 15, 2023 |
+
+
+**Compatibility matrix**
+
+|        | Cardano Node 1.35.7 | Cardano Node 8.0.8 | Cardano Node 8.1.1 | Cardano Node 8.1.2 |
+| -------| ------------------| --------------| ------------| -----------|
+| `marlowe-cli-0.1.0.0` | ✗ | ✗ | ✗ | ✓ |
+| `marlowe-cli-0.0.12.0` | ✓ | ✓ | ✓ | ? |
+| `marlowe-cli-0.0.11.0` | ✓ | ✓ | ✓ | ? |
```
Comment (bwbush) [inline docs/platform-and-architecture/versioning-policy.md:41]: We verified that the old versions don't work with the new node:
```suggestion
| `marlowe-cli-0.0.12.0` | ✓ | ✓ | ✓ | ✗ |
| `marlowe-cli-0.0.11.0` | ✓ | ✓ | ✓ | ✗ |
```

**Review by bwbush — APPROVED** [review]

**Pull request merged**